core/src/main/java/io/questdb/log/LogRollingFileWriter.java

```diff
@@ -394,8 +394,9 @@ private void removeExcessiveLogs() {
         // The list is sorted by last modification ts ASC, so we iterate in reverse order
         // starting with the newest files.
         long totalSize = 0;
-        for (long i = logFileList.size() - 1; i > -1; i -= 2) {
-            final long packedOffsets = logFileList.get(i);
+        // Leave last file on disk always.
+        for (long i = logFileList.size() - 3; i > 0; i -= 2) {
```

Comment:
nit: could be
```suggestion
        for (long i = logFileList.size() - 4; i > -1; i -= 2) {
```
and then
```java
final long packedOffsets = logFileList.get(i);
```

Reply:
I did this and a few more improvements